detectron2/data/datasets/coco.py

```diff
@@ -280,6 +287,131 @@ def file2id(folder_path, file_path):
     return dataset_dicts
 
 
+def convert_to_coco_dict(dataset_name):
+    """
+    Convert a generic dataset into COCO json format
+
+    Generic dataset description can be found here:
+    https://detectron2.readthedocs.io/tutorials/datasets.html#register-a-dataset
+
+    COCO data format description can be found here:
+    http://cocodataset.org/#format-data
+
+    Args:
+        dataset_name: name of the source dataset
+    Returns:
+        coco_dict: serializable dict in COCO json format
+    """
+
+    dataset_dicts = DatasetCatalog.get(dataset_name)
+    categories = [
+        {"id": id, "name": name}
+        for id, name in enumerate(MetadataCatalog.get(dataset_name).thing_classes)
+    ]
+
+    logger.info("Converting dataset dicts into COCO format")
+    coco_images = []
+    coco_annotations = []
+
+    # just for logging purposes
+    _annotation_keys = Counter()
+    for image_dict in dataset_dicts:
+        coco_image = {
+            "id": image_dict["image_id"],
+            "width": image_dict["width"],
+            "height": image_dict["height"],
+            "file_name": image_dict["file_name"],
+        }
+        coco_images.append(coco_image)
+
+        anns_per_image = image_dict["annotations"]
+        for annotation in anns_per_image:
+            # create a new dict with only COCO fields
+            coco_annotation = {}
+
+            # COCO requirement: XYWH box format
+            bbox = annotation["bbox"]
+            bbox_mode = annotation["bbox_mode"]
+            bbox = BoxMode.convert(bbox, bbox_mode, BoxMode.XYWH_ABS)
+
+            # COCO requirement: instance area
+            if "segmentation" in annotation:
+                # Computing areas for instances by counting the pixels
+                segmentation = annotation["segmentation"]
+                # TODO: check segmentation type: RLE, BinaryMask or Polygon
+                polygons = PolygonMasks([segmentation])
+                area = polygons.area()[0].item()
+            else:
+                # Computing areas using bounding boxes
+                area = Boxes([bbox]).area()[0].item()
+
+            # COCO requirement:
+            #   linking annotations to images
+            #   "id" field must start with 1
+            coco_annotation["id"] = len(coco_annotations) + 1
+            coco_annotation["image_id"] = image_dict["image_id"]
+            coco_annotation["bbox"] = bbox
+            coco_annotation["area"] = area
+            coco_annotation["category_id"] = annotation["category_id"]
+            coco_annotation["iscrowd"] = annotation["iscrowd"]
```

Comment:
Thanks for the PR! Just trying it out, I think it will be nice to make ``iscrowd`` field optional, and assume it's false by default - if someone is building their own dataset without "crowd" support, they likely won't have it.
```suggestion
            coco_annotation["iscrowd"] = annotation.get("iscrowd", 0)
```

Reply:
I love it!